### DISCUSSION TITLE
Support for migrating Kotlin application

### DISCUSSION THREAD

**Discussion by yong-at-zettle** [spring-projects-experimental/spring-boot-migrator#448] [Q&A] [answered]
Hi,
Is there a plan to support migrating Kotlin application? Thanks!

Cheers,
Yong

**Comment by fabapp2** [comment]
> Hi,
> 
> Is there a plan to support migrating Kotlin application? Thanks!
> 
> 
> 
> Cheers,
> 
> Yong

Hi @yong-at-zettle!
 
Thanks for your interest.
Unfortunatelly Gradle is not yet supported. It would be possible but require more OpenRewrite Recipes modifying the Gralde files to apply the same modifications for Gradle as currently offered by BuildFile and imlemented in OpenRewriteMavenBuildFile for Maven. These must should go as contributions into OpenRewrite and then be used in a OpenRewriteGradleBuildFile also implementing BuildFile. (...and solving roadblocks on the way)
In case you'd be interested to look into this and maybe contribute something, please let us know.

**Reply by yong-at-zettle** [reply]
Thanks for the reply, @fabapp2 !
Actually I intended to ask the support of migrating a project with Kotlin + Maven setup. But since OpenRewrite focuses on Java and spring-boot-migrator is based on OpenRewrite, hence it limits the support for Kotlin + Maven setup, right?
Cheers,
Yong

**Reply by fabapp2** [reply]
Sorry @yong-at-zettle!
It was from my mobile and I just realized your question was about Kotlin. 🤣 🙈

**Comment by fabapp2** [comment, marked as answer]
Unfortunatelly no, not yet.
As parsing and AST modifications rely solely on OpenRewrite this depends on the current Kotlin support in OpenRewrite. OpenRewrite has support for Kotlin but I am not sure about the coverage. Theoretically, it means the same as with Gradle support but `JavaSource`  as base-interface and something like `OpenRewriteKotlinSource` as implementation. And most likely some roadblocks I am naively ignoring now.

**Reply by yong-at-zettle** [reply]
Good to know. Thanks, @fabapp2 !

Cheers,
Yong
